src/databricks/labs/ucx/hive_metastore/udfs.py

```diff
@@ -96,27 +107,40 @@ def _collect_tasks(self, catalog, database) -> Iterable[partial[Udf | None]]:
     def _describe(self, catalog: str, database: str, udf: str) -> Udf | None:
         """Fetches metadata like udf type, input, returns, data access and body
         if specified for a specific udf within the given catalog and database.
+        The output is different between SQL UDFs and Hive UDFs. Hive UDFs only returns the class & usage.
         """
         full_name = f"{catalog}.{database}.{udf}"
+        logger.debug(f"[{full_name}] fetching udf metadata")
+        describe = {}
+        current_key = ""
         try:
-            logger.debug(f"[{full_name}] fetching udf metadata")
-            describe = {}
-            for key_value in self._fetch(f"DESCRIBE FUNCTION EXTENDED {escape_sql_identifier(full_name)}"):
-                if ":" in key_value:  # skip free text configs that don't have a key
-                    key, value = key_value.split(":")
-                    describe[key] = value.strip()
+            for row in self._fetch(f"DESCRIBE FUNCTION EXTENDED {escape_sql_identifier(full_name)}"):
+                key_value = row["function_desc"]
+                if ":" in key_value:
+                    current_key, value = key_value.split(":", 1)
+                    describe[current_key] = value.strip()
+                elif current_key != "":  # append multiline returns, e.g. Input
+                    describe[current_key] += f"\n{key_value.strip()}"
             return Udf(
                 catalog=catalog.lower(),
                 database=database.lower(),
                 name=udf.lower(),
-                func_type=describe.get("Type", "UNKNOWN"),
+                func_type=describe.get("Type", describe.get("Class", "UNKNOWN")),
                 func_input=describe.get("Input", "UNKNOWN"),
                 func_returns=describe.get("Returns", "UNKNOWN"),
                 deterministic=describe.get("Deterministic", False),
-                data_access=describe.get("Type", "UNKNOWN"),
+                data_access=describe.get("Data Access", "UNKNOWN"),
                 comment=describe.get("Comment", "UNKNOWN"),
                 body=describe.get("Body", "UNKNOWN"),
             )
         except Exception as e:  # pylint: disable=broad-exception-caught
             logger.error(f"Couldn't fetch information for udf {full_name} : {e}")
             return None
+
+    @staticmethod
+    def _assess_udfs(udfs: Iterable[Udf]) -> Iterable[Udf]:
+        for udf in udfs:
+            if udf.func_type != "SCALAR":
```

Comment:
We still yield an UDF in the describe first, but it can't be migrated? Should that be a decision made in the crawler? Or in the code that migrates the udf

Reply:
the crawler will return all udfs found, but also assess the suitability/migrate-ability of them. Similar to other crawlers